### PR TITLE
Safe system deletes

### PR DESCRIPTION
**Pull request by semtexzv**

### CODE REVIEW

File: listener/upload.go [hunk 1]
```diff
@@ -333,14 +341,27 @@ func processUpload(account string, host *Host) (*models.SystemPlatform, error) {
...
+
+	var deleted models.DeletedSystem
+	if err := tx.Find(&deleted, "inventory_id = ?", host.ID).Error; err != nil &&
+		!gorm.IsRecordNotFoundError(err) {
+		return nil, errors.Wrap(err, "Checking deleted systems")
+	}
+
+	// If the system was deleted in last hour, don't register this upload
+	if deleted.InventoryID != "" && deleted.WhenDeleted.After(time.Now().Add(-DeletionThreshold)) {
+		utils.Log("inventoryID", host.ID).Info("Received recently deleted system")
```
Comment (Josca) [inline listener/upload.go:354]: any test for that case?

File: listener/upload.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ const (
 	ErrorProcessUpload     = "unable to process upload"
 	UploadSuccessNoEval    = "upload event handled successfully, no eval required"
 	UploadSuccess          = "upload event handled successfully"
+
+	DeletionThreshold = 4 * time.Hour
```
Comment (Josca) [inline listener/upload.go:29]: would be better to create env var for that with 4h default value

File: database_admin/migrations/020_delete_cache.up.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    when_deleted TIMESTAMP WITH TIME ZONE NOT NULL,
+    UNIQUE (inventory_id)
+) TABLESPACE pg_default;
+
+CREATE INDEX ON deleted_system (when_deleted);
+
+GRANT SELECT, INSERT, UPDATE, DELETE ON deleted_system TO listener;
+GRANT SELECT ON TABLE deleted_system TO evaluator;
+GRANT SELECT ON TABLE deleted_system TO manager;
+GRANT SELECT ON TABLE deleted_system TO vmaas_sync;
```
Comment (Josca) [inline database_admin/migrations/020_delete_cache.up.sql:14]: Maybe we should add delete permission to vmaas_sync and set up job to delete old deleted systems into vmaas_sync, right?

File: listener/upload.go (hunk 1 above)
Comment (semtexzv) [reply listener/upload.go:354]: Added test in delete_test.go

File: listener/upload.go (hunk 2 above)
Comment (semtexzv) [reply listener/upload.go]: Thanks, fixed

File: database_admin/migrations/020_delete_cache.up.sql (hunk 3 above)
Comment (semtexzv) [reply database_admin/migrations/020_delete_cache.up.sql:14]: I've gone with the way that vulnerability does it and perform cleanup of the deleted_system table in listener.

**Pull request merged**